controllers/metallb_controller.go

```diff
@@ -149,15 +157,36 @@ func (r *MetalLBReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return err
 	}
 
+	if r.EnvConfig.IsOpenshift {
+		return ctrl.NewControllerManagedBy(mgr).
+			For(&metallbv1beta1.MetalLB{}).
+			Watches(&openshiftapiv1.Network{}, &handler.EnqueueRequestForObject{}).
+			Complete(r)
+	}
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&metallbv1beta1.MetalLB{}).
 		Complete(r)
 }
 
-func (r *MetalLBReconciler) syncMetalLBResources(config *metallbv1beta1.MetalLB) error {
+func (r *MetalLBReconciler) syncMetalLBResources(ctx context.Context, config *metallbv1beta1.MetalLB) error {
 	logger := r.Log.WithName("syncMetalLBResources")
 	logger.Info("Start")
 
+	bgpType := params.BGPType(config, r.EnvConfig)
+	if r.EnvConfig.MustDeployFRRK8sFromCNO && r.EnvConfig.IsOpenshift && (bgpType == metallbv1beta1.FRRK8sExternalMode) {
+		supportsFRRK8s, err := openshift.SupportsFRRK8s(ctx, r.Client)
+		if err != nil {
+			return err
+		}
+		if !supportsFRRK8s {
+			return EmbeddedFRRK8sSupportNotAvailable
+		}
+
+		if err := openshift.DeployFRRK8s(ctx, r.Client); err != nil {
+			return err
+		}
+	}
+
 	err := config.Validate()
```

Comment:
should we add a validation for external + ns must be specified together?

Reply:
I guess, with the premise that with the current defaults, to get into this scenario the user must willingly remove the default env variable and shoot himself in the foot.